waku/v2/protocol/lightpush/waku_lightpush.go

```diff
@@ -336,6 +329,7 @@ func (wakuLP *WakuLightPush) Publish(ctx context.Context, message *wpb.WakuMessa
 	for _, peerID := range params.selectedPeers {
 		wg.Add(1)
 		go func(id peer.ID) {
+			params.requestID = protocol.GenerateRequestID()
```

Comment:
I wonder if it wouldnt be best to send the same requestID to all selected peers.
Also, it's not  big deal, but since `params` is a pointer, due to this being executed in a go routine, it's possible that we'll end up with a race condition here (not super problematic tbh since it's only the requestID being modified, and maybe can be solved with
```
paramsValue := *params 
paramsValue.requestID = protocol.GenerateRequestID()
defer wg.Done()
response, err := wakuLP.request(ctx, req, &paramsValue, id)
```
)

Reply:
good catch....will fix it. 
wrt requestID, it felt that it makes it easier for tracing...hence wanted to have separate one for each peer..